python/mim_control/robot_centroidal_controller.py

```diff
@@ -75,15 +74,17 @@ def compute_com_wrench(self, q, dq, des_pos, des_vel, des_ori, des_angvel):
 
         cur_angvel = arr(dq[3:6])
 
-        if self.rotate_vel_error:
-            # Rotate the des and current angular velocity into the world frame.
-            quat_des = pin.Quaternion(
-                des_ori[3], des_ori[0], des_ori[1], des_ori[2]
-            ).matrix()
-            des_angvel = quat_des.dot(des_angvel)
+        # Rotate the des and current angular velocity into the world frame.
+        quat_des = pin.Quaternion(
+            des_ori[3], des_ori[0], des_ori[1], des_ori[2]
+        ).matrix()
+        des_angvel = quat_des.dot(des_angvel)
 
-            quat_cur = pin.Quaternion(q[6], q[3], q[4], q[5]).matrix()
-            cur_angvel = quat_cur.dot(cur_angvel)
+        quat_cur = pin.Quaternion(q[6], q[3], q[4], q[5]).matrix()
+        cur_angvel = quat_cur.dot(cur_angvel)
+
+        # Rotate the base velocity into global frame as well.
+        vcom = quat_cur.dot(vcom)
```

Comment:
Before we had a flag `self.rotate_vel_error` which was by default off. Digging deeper into this, it turns out it should always be turned on but also needs to rotate the vcom from the base into the global frame (last line).

BIG NOTE: The python binding here uses the `q` and `dq` for the com approximation. In contrast, in the DG whole body controller I made the decision to use the actual com. Should we switch the python implementation to use the com by default as well?

Reply:
Yes we should, even better we need to replace the python implementation with the C++ binded one